diff --git a/src/shader_recompiler/specialization.h b/src/shader_recompiler/specialization.h
index fa14583a..814fab6e 100644
--- a/src/shader_recompiler/specialization.h
+++ b/src/shader_recompiler/specialization.h
@@ -52,10 +52,14 @@ struct ImageSpecialization {
     bool is_srgb = false;
     AmdGpu::CompMapping dst_select{};
     AmdGpu::NumberConversion num_conversion{};
-    // FIXME any pipeline cache changes needed?
     u32 num_bindings = 0;
 
-    bool operator==(const ImageSpecialization&) const = default;
+    bool operator==(const ImageSpecialization& other) const {
+        return type == other.type && is_integer == other.is_integer &&
+               is_storage == other.is_storage && is_cube == other.is_cube &&
+               is_srgb == other.is_srgb && dst_select == other.dst_select &&
+               num_conversion == other.num_conversion;
+    }
 };
 
 struct FMaskSpecialization {
